scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1864,13 +1865,22 @@ def matmat_XH_X(x):
 
         eigvals, eigvec = lobpcg(XH_X, X, tol=tol ** 2, maxiter=maxiter,
                                  largest=largest)
-
+    elif solver == 'propack':
+        from scipy.sparse.linalg import svdp
+        jobu = ((isinstance(return_singular_vectors, str) and 'u' in return_singular_vectors)
+                or (isinstance(return_singular_vectors, bool) and return_singular_vectors))
+        jobv = ((isinstance(return_singular_vectors, str) and 'v' in return_singular_vectors)
+                or (isinstance(return_singular_vectors, bool) and return_singular_vectors))
+        return svdp(
+            A, k=k, tol=tol**2, which=which, maxiter=maxiter, compute_u=jobu, compute_v=jobv,
+            irl_mode=False, kmax=ncv, v0=v0)
```

Comment:
```suggestion
            irl_mode=True, kmax=ncv, v0=v0)
```
You originally had this set `True`. I changed it to `False` in one of my PRs because `True` wasn't working, but that's fixed now. Do you know what the default should be?

For PROPACK, `v0` is supposed to be of length `A.shape[0]`, but for ARPACK, it's length `min(A.shape)`. 

Do you happen to know if the description of `ncv`:
> The number of Lanczos vectors generated ncv must be greater than k+1 and smaller than n; it is recommended that ncv > 2*k Default: min(n, max(2*k + 1, 20))

is perfectly appropriate for `kmax`? Or should we change the description?

On the other hand, I think it is ignored (along with several other options) when the solver is `lobpcg`. Really, maybe we need to document the different options separately as for `linprog`. I like "omnibus" functions, but I admit they are difficult to document properly when backends with different options are added.

Reply:
I believe `irl_mode` should be `True` because of this line in the docstring: "`which='SM'` requires `irl_mode=True`."  (Whether or not that statement is true, I haven't tested -- it's what came prewritten from the pypropack repo).

`kmax` will probably need to be looked into.  PROPACK calls `kmax` the "maximal dimension of the Krylov subspace" which could be the number of Lanczos vectors?  Minimum value of `kmax` for PROPACK is `5*k`